tests/sles4sap/redirection_tests/ensa2_kill_sapinstance.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Test module performs ENSA2 B<'Kill sapinstance'> test scenario.
+
+use parent 'sles4sap::sap_deployment_automation_framework::basetest';
+
+use warnings;
+use strict;
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use utils qw(script_retry);
+use hacluster;
+use sles4sap::sap_host_agent qw(saphostctrl_list_instances);
+use sles4sap::sapcontrol;
+use sles4sap::console_redirection;
+
+=head1 SYNOPSIS
+
+Test module performs ENSA2 B<'Kill sapinstance'> test scenario. SAP instance process (ERS or ASCS) B<*.sap*> is killed
+using C<kill -9> command.
+Test module is based on usage of console redirection. Check B<tests/sles4sap/redirection_tests/README.md> for details.
+
+Test module requires following structure containing information:
+    $run_args->{scenarios} = [
+        'GottaCatchEmAll' => {
+            description => 'Test throws pokeball at Pikachu and captures it.',
+            connect_ip => 192.168.1.1,
+            connect_user => Ash,
+            resource_name => pokeball
+        }
+    ];
+
+=cut
+
+sub run {
+    my ($self, $run_args) = @_;
+    my %scenario = %{$run_args->{scenarios}{$self->{name}}};
+    my $resource_name = $scenario{resource_name};
+
+    record_info('TEST INFO', $scenario{description});
+
+    connect_target_to_serial(
+        destination_ip => $scenario{connect_ip}, ssh_user => $scenario{connect_user}, switch_root => 1);
+
+    my @instances_data = @{saphostctrl_list_instances(as_root => 'yes', running => 'yes')};
+    my $instance_id = $instances_data[0]->{instance_id};
+    my $instance_type = get_instance_type(local_instance_id => $instance_id);
+
+    # Show status
+    sap_show_status_info(cluster => 1, netweaver => 1, instance_id => $instance_id);
+
+    # Check if cluster is being healthy
+    my $fail_count;
+    record_info('Cluster wait', 'Waiting for resources to start');
+    wait_until_resources_started();
+    wait_for_idle_cluster();
+
+    # Remove meta-argument 'migration-threshold' for cluster to try restarting sapinstance process first.
+    crm_resource_meta_set(resource => $resource_name, meta_argument => 'migration-threshold');
```

Comment:
How does this work? In the end runs `crm resource meta $resource_name delete migration-threshold`, but when is it added again to test the scenario when the resource is moved? Future code?

Reply:
Yep. There will be test which requires ASCS to migrate immediately, so the future code will:

1. Check and store current value
2. Change the value accordingly if failover is supposed to happen
3. Restore value at the end of test module

Does it make sense?

Reply:
> Does it make sense?

Yes. Perfectly. Thanks.